src/sfall_metarules.cc

```diff
@@ -567,6 +572,57 @@ void sprintf_lite(Program* program, int args, const char* infoOpcodeName)
     programStackPushString(program, out);
 }
 
+// message_box
+void mf_message_box(Program* program, int args)
+{
+    static int dialogShowCount = 0;
+
+    const char* string = programStackPopString(program);
+    if (string == nullptr || string[0] == '\0') {
+        return;
+    }
+
+    char* copy = internal_strdup(string);
+
+    const char* body[4];
+    int count = 0;
+
+    char* pch = strchr(copy, '\n');
+    while (pch != nullptr && count < 4) {
+        *pch = '\0';
+        body[count++] = pch + 1;
+        pch = strchr(pch + 1, '\n');
+    }
+
+    int flags = DIALOG_BOX_LARGE | DIALOG_BOX_YES_NO;
+    if (args > 1) {
+        int flagParam = programStackPopInteger(program);
+        if (flagParam != -1) {
+            flags = flagParam;
+        }
+    }
+
+    // note: most of the CE code uses colorTable indices, but this metarule expects palette values.
+    // Default: yellow (145) = _colorTable[32328]
+    int color1 = _colorTable[32328], color2 = _colorTable[32328];
+    if (args > 2) {
+        color1 = programStackPopInteger(program);
+    }
+    if (args > 3) {
+        color2 = programStackPopInteger(program);
+    }
+
+    dialogShowCount++;
+    scriptsDisable();
```

Comment:
this copies the sfall impl.  Script must be disabled since otherwise it'll keep running while the dialog is open, and get a stack underflow.